usermods/TetrisAI_v2/gridbw.h

```diff
@@ -112,6 +112,17 @@ class GridBW
     {
         return pixels[y] == (uint32_t)((1 << width) - 1);
     }
+
+    void reset()
+    {
+        if (width > 32)
+        {
+            throw std::invalid_argument("maximal width is 32");
```

Comment:
throw & catch are not supported on 8266. So this "thow" could cause a crash.

Reply:
Ok what is your suggestion?
What would be a good alternative?

Reply:
Btw, why is the max width 32 pixels? Just lack of testing, or math breaks down (overflows), or too much memory needed? We have users with very large fixtures...

If your effect has some limits, the normal way is `return mode_static();`

Reply:
The internal engine works on uint32 rows (32 pixels of max. width).
This way the grid has this limitation.

So I will return mode_static(); then.

Reply:
I clamped the value to 32 and added a section to the readme

Reply:
👍

Reply:
Looks like you still need to change the second "throw" in line 120

Reply:
indeed. fixed.

Reply:
One more thing regarding the size. If the segment is larger than 32 pixels, just center everything and leave black bars on the sides.

Reply:
You can, but that is a bad UX.
With centered game and leaving black bars, you are hinting a user that there is nothing of interest on the sides (or top & bottom) while still having a meaningful display.

Reply:
You need to calculate offset manually.
```c++
int offsetX = cols > 32 ? (cols - width) / 2 : 0;
int offsetY = rows > 32 ? (rows - height) / 2 : 0;
...
SEGMENT.setPixelColorXY(x + offsetX, y + offsetY, color);
```
Where `width` and `height` are your desired/calculated width/height of the effect display.

Reply:
Yes thank you. In the meantime, I have already solved it in exactly the same way.

Reply:
I think it should have been resolved but while testing I found that 'peek' seems to scale pixels 'down' if I try to define something like 36 x 260.

Is there a way to simulate WLED (eg. vitual matrix, X86 simulator, debug window, etc.)?

Reply:
The effect gets centered now. The maximal height gets also checked now. The 'next piece' and the border are placed smarter. You are right it was worth to have another look.